### PR TITLE
Fixing CCScale9Sprites when loaded from CCB files

### PR DESCRIPTION
**Pull request by Clarinexus**
CCScale9Sprites that were loaded from CCB files could not be set to a color or opacity other than white and full opacity. This fix allows CCB-loaded sprites to have dynamic color and opacity again.

### CODE REVIEW

**Comment by dumganhar** [comment]
Yep, your fix is right.

> CCScale9Sprites that were loaded from CCB files could not be set to a color or opacity other than white and full opacity. 

But the _setColor_ and _setOpacity_ did work. Why did you say that `could not be set to a color or opacity other than white and full opacity.`?

**Comment by Clarinexus** [comment]
The functions setColor and setOpacity worked only for programmatically-added CCScale9Sprites, but ones that were loaded in through CCB files couldn't be changed from white or fully opaque, and weren't taking the CCB file's color/opacity settings on creation.

**Comment by Clarinexus** [comment]
The real problem was that updateWithBatchNode was calling getColor and getOpacity and then setting that color/opacity at the end, causing CCB-loaded sprites to go white/opaque.

File: extensions/GUI/CCControlExtension/CCScale9Sprite.h [hunk 1]
```diff
@@ -308,7 +308,9 @@ class CCScale9Sprite : public CCNodeRGBA
      */
     virtual bool isOpacityModifyRGB(void);
     virtual void setOpacity(GLubyte opacity);
+	virtual GLubyte CCScale9Sprite::getOpacity();
```
Comment (dumganhar) [inline extensions/GUI/CCControlExtension/CCScale9Sprite.h:311]: `CCScale9Sprite::` should be removed.

File: extensions/GUI/CCControlExtension/CCScale9Sprite.h [hunk 2]
```diff
@@ -308,7 +308,9 @@ class CCScale9Sprite : public CCNodeRGBA
...
     virtual void setColor(const ccColor3B& color);
+	virtual const ccColor3B& CCScale9Sprite::getColor();
```
Comment (dumganhar) [inline extensions/GUI/CCControlExtension/CCScale9Sprite.h:313]: CCScale9Sprite:: should be removed.

**Pull request merged**